### PR TITLE
GRO-1563: Create endpoint and basic models

### PR DESCRIPTION
**Pull request by DanielFernandesPTech**

### CODE REVIEW

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **danielfernandesptech**
You can assign the PR to them by writing `/assign @danielfernandesptech` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-155** [here](http://ms-lending-api-pr155.selinadev.co.uk)

File: src/main/java/com/selina/lending/api/controller/QuickQuoteController.java [hunk 1]
```diff
@@ -52,6 +60,15 @@ public ResponseEntity<QuickQuoteResponse> createQuickQuoteApplication(
         return ResponseEntity.ok(filterQuickQuote(quickQuoteApplicationRequest));
     }
 
+    @Override
+    @Permission(resource = QQ_CC, scope = Create)
+    public ResponseEntity<QuickQuoteResponse> createQuickQuoteCCApplication(
+            QuickQuoteApplicationRequest quickQuoteApplicationRequest) {
+        log.info("Create Quick Quote CC application with [externalApplicationId={}]", quickQuoteApplicationRequest.getExternalApplicationId());
+        var x = filterQuickQuoteCC(quickQuoteApplicationRequest);
```
[the remark this answers is not in the document]
Comment (DanielFernandesPTech) [reply src/main/java/com/selina/lending/api/controller/QuickQuoteController.java:68]: It's true. I do that for debug and check a value of this variable. My Bad xD

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-155** [here](http://ms-lending-api-pr155.selinadev.co.uk)

File: src/main/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapper.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package com.selina.lending.internal.mapper;
+
+
+import com.selina.lending.internal.dto.quote.QuickQuoteApplicationRequest;
+import com.selina.lending.internal.service.application.domain.quotecc.QuickQuoteCCRequest;
+import org.mapstruct.Mapper;
+import org.mapstruct.factory.Mappers;
+
+@Mapper(uses = {QuickQuoteApplicantMapper.class, LoanInformationMapper.class, QuickQuotePropertyDetailsMapper.class})
+public interface QuickQuoteCCRequestMapper {
```
Comment (KastorDriver) [inline src/main/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapper.java:10]: Shall we move all mappers that are specific to QQ_CC to the separate package `*.mappers.quotecc` ?

File: src/test/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import com.selina.lending.internal.service.application.domain.quotecc.QuickQuoteCCRequest;
+import org.junit.jupiter.api.Test;
+
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class QuickQuoteCCRequestMapperTest extends MapperBase{
```
Comment (KastorDriver) [inline src/test/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapperTest.java:12]: Shall we move all mappers tests that are specific to QQ_CC to the separate package `*.mapper.quotecc` ?

File: src/main/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapper.java (hunk 2 above)
Comment (DanielFernandesPTech) [reply src/main/java/com/selina/lending/internal/mapper/QuickQuoteCCRequestMapper.java:10]: If we do that maybe the project organization will be more simply and understandable. What you think about?

**Comment by selina-bot-user** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[KastorDriver](https://github.com/Selina-Finance/ms-lending-api/pull/155#issuecomment-1494461260 "Approved")*

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=Selina-Finance%2Fms-lending-api).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/Selina-Finance/ms-lending-api/blob/main/OWNERS)~~ [KastorDriver]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by selina-bot-user** [comment]
:star: PR built and available in a preview **selina-finance-ms-lending-api-pr-155** [here](http://ms-lending-api-pr155.selinadev.co.uk)

**Pull request merged**